xbmc/music/windows/GUIWindowVisualisation.cpp

```diff
@@ -240,3 +241,29 @@ void CGUIWindowVisualisation::FrameMove()
   }
   CGUIWindow::FrameMove();
 }
+
+void CGUIWindowVisualisation::ToggleOSD()
+{
+  CGUIDialogMusicOSD *pOSD = static_cast<CGUIDialogMusicOSD *>(g_windowManager.GetWindow(WINDOW_DIALOG_MUSIC_OSD));
+  if (pOSD)
+  {
+    if (pOSD->IsDialogRunning())
+    {
+      pOSD->Close();
+    }
+    else if (g_infoManager.GetShowInfo() &&
+             CSettings::Get().GetBool("pvrplayback.confirmchannelswitch") &&
+             g_application.CurrentFileItem().HasPVRChannelInfoTag())
+    {
+      // If only channel info but no player controls are currently visible
+      // and setting "confirm channel switch with OK" is active, do a channel
+      // switch instead of turning on the the player controls.
+      g_application.m_pPlayer->SwitchChannel(g_application.CurrentFileItem().GetPVRChannelInfoTag());
```

Comment:
no it won't work, because with my remote.xml you will end-up with the same issue. I can't confirm the channel switch with "OK" on my remote, because it sends `ACTION_SHOW_PLAYLIST`.